### PR TITLE
OCPCLOUD-2643: MachineSetSync controller bootstrap

### PR DESCRIPTION
**Pull request by theobarberbany**
-> Bootstraps MachineSet Sync controller
-> Adds basic test suite & empty test.
-> Fetches resources + dynamic watching of InfraMachineTemplates

### CODE REVIEW

File: pkg/util/watch_filters.go [hunk 1]
```diff
@@ -43,6 +46,60 @@ func CAPIMachineToMAPIMachine(namespace string) func(context.Context, client.Obj
 	}
 }
 
+// RewriteNamespace takes a client.Object and returns a reconcile.Request for
+// it in the namespace provided.
+//
+// It is intended for use with CAPI Machines and MachineSet requests, where we expect
```
Comment (theobarberbany) [inline pkg/util/watch_filters.go:52]: I think CAPIMachineToMAPIMachine above should be replaced by RewriteNamespace, however I'm not doing that just yet as I want https://github.com/openshift/cluster-capi-operator/pull/212 to merge first, which relies on the other naming.

File: pkg/controllers/machinesetsync/machineset_sync_controller.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+const (
+	capiNamespace string = "openshift-cluster-api"
+	mapiNamespace string = "openshift-machine-api"
+)
+
+var (
+	// errPlatformNotSupported is returned when the platform is not supported.
+	errPlatformNotSupported = errors.New("error determining InfraMachineTemplate" +
+		" type, platform not supported")
```
Comment (JoelSpeed) [inline pkg/controllers/machinesetsync/machineset_sync_controller.go:46]: Why split this over two lines?

File: pkg/controllers/machinesetsync/machineset_sync_controller.go [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+	// default to the production namespaces.
+	if r.CAPINamespace == "" {
+		r.CAPINamespace = capiNamespace
+	}
+
+	if r.MAPINamespace == "" {
+		r.MAPINamespace = mapiNamespace
+	}
+
+	// TODO: InfraMachineTemplates
```
Comment (JoelSpeed) [inline pkg/controllers/machinesetsync/machineset_sync_controller.go:76]: Still TODO?

File: pkg/util/watch_filters.go (hunk 1 above)
Comment (JoelSpeed) [reply pkg/util/watch_filters.go:52]: 212 is now merged so you should be able to fix this up now

File: pkg/controllers/machinesetsync/machineset_sync_controller.go (hunk 2 above)
Comment (theobarberbany) [reply pkg/controllers/machinesetsync/machineset_sync_controller.go:46]: Ah - I was thinking about trying to keep everything wrapped to ~80 characters, but don't think it's worth it.

**Review by JoelSpeed — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**